sway-lib-std/src/inputs.sw

```diff
@@ -0,0 +1,220 @@
+//! Getters for fields on transaction inputs.
+//! This includes InputCoins, InputMessages and InputContracts.
+library inputs;
+
+use ::address::Address;
+use ::mem::read;
+use ::option::Option;
+use ::revert::revert;
+use ::tx::{
+    tx_type,
+    Transaction,
+    GTF_SCRIPT_INPUT_AT_INDEX,
+    GTF_CREATE_INPUT_AT_INDEX,
+    GTF_SCRIPT_INPUTS_COUNT,
+    GTF_CREATE_INPUTS_COUNT,
+};
+
+const GTF_INPUT_TYPE = 0x101;
+
+////////////////////////////////////////
+// GTF Opcode const selectors
+////////////////////////////////////////
+
+const GTF_INPUT_COIN_TX_ID = 0x102;
+const GTF_INPUT_COIN_OUTPUT_INDEX = 0x103;
+const GTF_INPUT_COIN_OWNER = 0x104;
+const GTF_INPUT_COIN_AMOUNT = 0x105;
+// const GTF_INPUT_COIN_ASSET_ID = 0x106;
+// const GTF_INPUT_COIN_TX_POINTER = 0x107;
+// const GTF_INPUT_COIN_WITNESS_INDEX = 0x108;
+// const GTF_INPUT_COIN_MATURITY = 0x109;
+// const GTF_INPUT_COIN_PREDICATE_LENGTH = 0x10A;
+// const GTF_INPUT_COIN_PREDICATE_DATA_LENGTH = 0x10B;
+// const GTF_INPUT_COIN_PREDICATE = 0x10C;
+const GTF_INPUT_COIN_PREDICATE_DATA = 0x10D;
+
+const GTF_INPUT_CONTRACT_TX_ID = 0x10E;
+const GTF_INPUT_CONTRACT_OUTPUT_INDEX = 0x10F;
+// const GTF_INPUT_CONTRACT_BALANCE_ROOT = 0x110;
+// const GTF_INPUT_CONTRACT_STATE_ROOT = 0x111;
+// const GTF_INPUT_CONTRACT_TX_POINTER = 0x112;
+// const GTF_INPUT_CONTRACT_CONTRACT_ID = 0x113;
+
+// const GTF_INPUT_MESSAGE_MESSAGE_ID = 0x114;
+// const GTF_INPUT_MESSAGE_SENDER = 0x115;
+// const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
+const GTF_INPUT_MESSAGE_AMOUNT = 0x117;
+// const GTF_INPUT_MESSAGE_NONCE = 0x118;
+const GTF_INPUT_MESSAGE_OWNER = 0x119;
```

Comment:
The latest [spec changes](https://github.com/FuelLabs/fuel-specs/commit/5dcf71f162b11fb9ce998315a7078630d78536b7#diff-0ef30bdacf4ca9f9f500f9aaf4a50326591005bb730a703d5b8abc049bb475e7) removes the `owner` field from `InputMessages` and some of these constants have also updated to remove the gap. If [this issue](https://github.com/FuelLabs/fuel-asm/issues/92) is already closed, then those updates will need to be made here as well. Otherwise we will need to create a separate issue for the `owner` field on `InputMessage`

Reply:
Ok, good to know @pixelcircuits.
I will create an issue to track this and try to keep moving forward with this as is for now until a new fuel-core is released with the changes removing the owner field. Otherwise I'm aiming for a moving target with this PR and it will never be merged.